### ISSUE TITLE
[GENERAL] Birthdate system keeps asking me to grow up

### ISSUE THREAD

**Issue by TheToaster98** [VOREStation/VOREStation#15141]
#### Brief description of the bug
When a character's birthday is past due, it will keep asking if I want to age up each single time I spawn on that character.
Selecting either option will up my age or not correctly, but I will still get asked next time I spawn.

This can even happen in the same round, if I leave and rejoin.

#### What you expected to happen
> Do you want to age up? This will only show up once per year.

Select option.
The end

#### What actually happened
> Do you want to age up? This will only show up once per year.
Yes

Next round:
> Do you want to age up? This will only show up once per year.
No.

Next Round:

> Do you want to age up? This will only show up once per year.
Please leave me alone.

Next round:

> Do you want to age up? This will only show up once per year.

#### Steps to reproduce
- Set birthday to day before current date.
- Join.
- Select either option.
- Leave.
- Rejoin.
- Popup happens again.

#### Code Revision
[2e9c5578da693d25fe5f1552e9a890d0d638ec42](http://127.0.0.1:59018/tmp11900/%3Ca%20href=%22https://github.com/VOREStation/VOREStation/commit/2e9c5578da693d25fe5f1552e9a890d0d638ec42%22%3Ehttps://github.com/VOREStation/VOREStation/commit/2e9c5578da693d25fe5f1552e9a890d0d638ec42%3C/a%3E)

**Comment by Very-Soft** [comment]
Tested on live server using both yes and no for whether or not to age up, deleted the character, respawned them, got no notification. could not replicate issue

**Comment by KillianKirilenko** [comment]
I'm also experiencing this. A possible notable factor is that the character I'm getting it on was made *prior* to the addition of the birthday system?

**Comment by TheToaster98** [comment]
> 

Exactly the situation at hand, character made prior to the system addition, using TGUI.
I add pictures taken in the main server as of five minutes ago, after selecting the "I want to stay young" option and leaving, rejoining prompted the same dialogue once more.
![Screenshot 2023-08-02 154648](https://github.com/VOREStation/VOREStation/assets/51209769/9de99dce-ae2b-4906-8b75-6f78a5e6ea35)
![Screenshot 2023-08-02 154704](https://github.com/VOREStation/VOREStation/assets/51209769/1050df03-0f5a-4613-b5a7-a2f2a6feb9e8)
I apologise for not answering to Soft's comment, I don't check my mail very often during Summer.

**Comment by Heroman3003** [comment]
Is this issue still present?

**Comment by KillianKirilenko** [comment]
It's stopped occurring on my end. I believe what might have fixed it was switching to old-style prompts before hitting the "don't change age" button, but I can't confirm that easily.

**Comment by TheToaster98** [comment]
Due to my PSU having caught on fire I won't be capable of testing this, please wait patiently.

**Comment by TheToaster98** [comment]
Decided to wait for a different character to pass their birthday, since it was a week away from my last post.
This character was also made before the system was implemented.

I waited for the birthday date (yesterday) and accepted the level up, which did indeed change the age, upon joining the next day (today), I was prompted to level up again.
So I denied aging while TGUI input was on, left.
Turned TGUI input off and rejoined, selected to not age.
Left again, rejoined once more... _Do you want to level up?_
I haven't tried editing the age directly in the setup, but I have the feeling that even if it makes the popup go away it may come back in a year, which doesn't fix much if the point of the system is to not have to change it manually.

TL;DR: Different char, same issue, TGUI unrelated.

**Comment by TheToaster98** [comment]
Update to the update to the update.
Bug no longer happening.
Solution: Pressing reconnect while on an affected character will cause the popup to show up once more right after you reconnect. Selecting either option will stop the pop up showing up on consequent spawns.

Still broken though.

**Comment by TheToaster98** [comment]
Poking this because it is still happening and I won't let this bug be forgotten.

**Comment by TheToaster98** [comment]
I am not letting Stalebot take it away to the shadow realm. Happy belated birthday :D

**Comment by Kashargul** [comment]
Should be resolved?

**Comment by TheToaster98** [comment]
> Should be resolved?

You won't believe it.
Still broken.
Method of reproduction:
- Changed date of birth to today, spawned in, selected not to age up.
- Jumped into the teleporter, ghosted, waited for char to be deleted.
- Joined back as the same character, pop up happens again.

Extra details:

I have all the TGUI stuff on.

**Comment by Kashargul** [comment]
> > Should be resolved?
> 
> You won't believe it. Still broken. Method of reproduction:
> 
>     * Changed date of birth to today, spawned in, selected not to age up.
> 
>     * Jumped into the teleporter, ghosted, waited for char to be deleted.
> 
>     * Joined back as the same character, pop up happens again.
> 
> 
> Extra details:
> 
> I have all the TGUI stuff on.

can you retest this?
